instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -437,7 +439,8 @@ def _teardown_request(exc):
             return
 
         activation = flask.request.environ.get(_ENVIRON_ACTIVATION_KEY)
-        if not activation:
+        thread_id = flask.request.environ.get(_ENVIRON_THREAD_ID_KEY)
+        if not activation or thread_id != get_ident():
             # This request didn't start a span, maybe because it was created in
             # a way that doesn't run `before_request`, like when it is created
             # with `app.test_request_context`.
```

Comment:
can we update this comment? We now have another condition that causes us to not call `__exit__`

Reply:
I think this could be the only/sole place that we have an in-code comment about what we're doing with all these threading shenanigans.

And I do think recording + comparing against the `get_ident()` is weird enough that it's worth documenting in code a bit via a comment somewhere

Reply:
Good call. Updated the comment.

Reply:
🎉 comment looks great, thanks